core/src/test/java/co/cask/wrangler/steps/transformation/ExpressionTest.java

```diff
@@ -169,5 +181,48 @@ public void testMethodCalls() throws Exception {
     records = PipelineTest.execute(directives, records);
     Assert.assertTrue(records.size() == 0);
   }
+
+  @Test
+  public void testGeoFence() throws Exception {
+
+    String geoJsonFence = "{\"type\":\"FeatureCollection\",\"features\":[{\"type\":\"Feature\",\"properties\":{}," +
+        "\"geometry\":{\"type\":\"Polygon\",\"coordinates\":[[[-122.05870628356934,37.37943348292772]," +
+        "[-122.05724716186525,37.374727268782294],[-122.04634666442871,37.37493189292912]," +
+        "[-122.04608917236328,37.38175237839049],[-122.05870628356934,37.37943348292772]]]}}]}";
+
+    String[] directives = new String[]{
+        "set column result geo:inFence(lat,lon,fences)"
+    };
+
+    List<Record> records = Arrays.asList(
+        new Record("id", 123)
+            .add("lon", -462.49145507812494)
+            .add("lat", 43.46089378008257)
+            .add("fences", geoJsonFence)
+    );
+    records = PipelineTest.execute(directives, records);
+    Assert.assertFalse((Boolean) records.get(0).getValue("result"));
+  }
+
+  @Test(expected = StepException.class)
```

Comment:
No, The exception is thrown while forming the Jexl expression for that particular step
https://github.com/hydrator/wrangler/blob/develop/core/src/main/java/co/cask/wrangler/steps/transformation/Expression.java#L115

IMO logically also, since each row will have its own value of Json, the exception is specific to the step